code/modules/mob/living/silicon/pai/pai.dm

```diff
@@ -163,22 +151,41 @@
 	if(delold)
 		qdel(src)
 
+/datum/action/innate/pai
+	name = "PAI Action"
+	var/mob/living/silicon/pai/P
+
 /datum/action/innate/pai/Trigger()
-	if(ispAI(owner))
-		var/mob/living/silicon/pai/P = owner
-		P.pai_ui_action(src)
+	if(!ispAI(owner))
+		return 0
+	P = owner
 
 /datum/action/innate/pai/shell
 	name = "Toggle Holoform"
 	button_icon_state = "pai_holoform"
 	background_icon_state = "bg_tech"
 
+/datum/action/innate/pai/shell/Trigger()
+	..()

```

Comment:
if you like want . = ..() and if(.), so the owner check is properly managed for each child